### PR TITLE
Specs for views and fix n+1

### PR DESCRIPTION
**Pull request by MudasirSherwani**
## We the team have done the required job in the branch:

**- Fix N+1 Problem**

#### Before:
Screenshot 2023-04-12 172201

#### After:
Screenshot 2023-04-12 173404

- Add Test for Users
- Add test for Posts

### CODE REVIEW

File: spec/features/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  before(:each) do
+    User.create(id: 1, name: 'Mudasir Sherwani', post_counter: 5, photo: 'https://i.imgur.com/1J3wZQx.jpg',
+                bio: 'I am Mudasir Sherwani the full stack developer')
+    Post.create(title: 'Post #1', text: 'This is my first post', comment_counter: 0,
+                like_counter: 0, author_id: 1)
+    Like.create(author_id: 1, posts_id: 1)
+  end
+  it 'renders the title of the post' do
+    get user_posts_path(1)
+    expect(response.body).to include('Post #1')
```
Comment (VICTOR-CHUKWU) [inline spec/features/post_index_spec.rb]: - [x] this test is currently failing, kindly fix it 
image

File: spec/features/user_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    expect(response.body).to include('Tom', 'Lilly')
+  end
+  it 'shows number of posts of user' do
+    get users_path
+    expect(response.body).to include('<p>Number of posts: 2</p>')
+  end
+  it 'renders the profile picture of the users' do
+    get users_path
+    expect(response.body).to include('img')
+  end
```
Comment (VICTOR-CHUKWU) [inline spec/features/user_index_spec.rb]: - [x] kindly add a test that redirects to the user's show page when you click on a user.

File: spec/features/user_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    get user_path(1)
+    expect(response.body).to include('A Full Stack Developer')
+  end
+  it 'renders the number of posts of the user' do
+    get user_path(1)
+    expect(response.body).to include('5')
+  end
+  it 'renders the profile picture of the user' do
+    get user_path(1)
+    expect(response.body).to include('img')
```
Comment (VICTOR-CHUKWU) [inline spec/features/user_show_spec.rb]: - [x] kindly add Tests that redirect to the post's show page when you click on a user's post and test that redirects to the user's post's index page when you click on a **see all posts**

File: spec/features/post_index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    expect(response.body).to include('0')
+  end
+  it 'renders the number of comments of the post' do
+    get user_posts_path(1)
+    expect(response.body).to include('0')
+  end
+  it 'renders the text of the post' do
+    get user_posts_path(1)
+    expect(response.body).to include('This is my first post')
+  end
```
Comment (VICTOR-CHUKWU) [inline spec/features/post_index_spec.rb]: - [x] I believe this is a file if for ` post show` but the file name is `post_index`, kindly change that as it is confusing. also kindly add `tests that asserts you can see who wrote the post, that asserts you can see the username of each commenter that asserts you can see the comment each commentor left`.
- [x] also I can not see tests for `post-index` kindly add that in a separate file

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @MudasirSherwani  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/post_index_spec.rb (hunk 1 above)
Comment (MudasirSherwani) [reply spec/features/post_index_spec.rb]: Sir, there is no failed test on my side:
Suggest to me how can I fix it.
Thank you.
Screenshot 2023-04-12 202759

File: spec/features/user_index_spec.rb (hunk 2 above)
Comment (Abenezer-Tilahun) [reply spec/features/user_index_spec.rb]: Fixed.

File: spec/features/user_show_spec.rb (hunk 3 above)
Comment (Abenezer-Tilahun) [reply spec/features/user_show_spec.rb]: Fixed.

File: spec/features/post_index_spec.rb (hunk 4 above)
Comment (Abenezer-Tilahun) [reply spec/features/post_index_spec.rb]: Fixed.

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @MudasirSherwani 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**